Black Clovedle/ViewController.swift

```diff
@@ -6,14 +6,710 @@
+                   UserDefaults.standard.set(allTimeCorrect, forKey: "All-Time Correct")
+                   UserDefaults.standard.set(currentDate, forKey: "lastUpdatedDate")
+               } else {
+                   characterImagesArray.append(characterObject!.imageName!)
+               }
+           }
+    }
+    
+    
+    
+    
+    
+    func updateCurrentDate() {
+        let df = DateFormatter()
+        df.dateFormat = "MM/dd"
+        currentDate = df.string(from: Date())
+        if currentGameDate != currentDate {
+            clearGuessTable()
+        }
+    }
+    
+    
+    
+    func getRandomCharacter() {
+         let randomCharacter = getNameForToday()
+            print("Random Character: \(randomCharacter.name), Gender: \(randomCharacter.gender) Affiliation: \(randomCharacter.affiliation), Magic Attribute: \(randomCharacter.magicAttribute), Debut Arch: \(randomCharacter.debutArc), Arch Number: \(randomCharacter.debutArc)")
+            correctName = randomCharacter.name
+            correctGender = randomCharacter.gender
+            correctAffiliation = randomCharacter.affiliation
+            correctMagicAttribute = randomCharacter.magicAttribute
+            correctDebutArch = randomCharacter.debutArc
+            correctSpirit = randomCharacter.spirit
+            correctImage = randomCharacter.imageName
+            correctArchNumber = randomCharacter.arcNumber
+    }
+    
+    
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        if segue.identifier == "CharacterCorrectViewController" {
+            if let destinationVC = segue.destination as? CharacterCorrectViewController {
+                destinationVC.modalPresentationStyle = .fullScreen
+                destinationVC.correctCharacter = correctName!
+                print("this is the print in the view controller \(guesses.count)")
+                destinationVC.guessesCount = guesses.count
+                destinationVC.correctImageSecondView = UIImage(named: "\(correctName!)")
+            }
+        }
+    }
+    
+    func resetGame(){
+        getRandomCharacter()
+        guessText.text = ""
+        tryAgain.text = "Guess today's Black Clover Character!"
+        //        tryAgain.text = correctName
+        numberOfGuesses = 0
+        guesses = []
+        gender = [UIImage]()
+        affiliation = [UIImage]()
+        magicAttribute = [UIImage]()
+        debutArch = [UIImage]()
+        spirit = [UIImage]()
+        genderTextArray = [String]()
+        affiliationTextArray = [String]()
+        magicTextArray = [String]()
+        spiritTextArray = [String]()
+        debutArchTextArray = [String]()
+        characterImagesArray = [UIImage]()
+        self.tableView.reloadData()
     }
+}
+
+extension ViewController: UITableViewDataSource{
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        if tableView == self.tableView {
+            return numberOfGuesses
```

Comment:
Similar to the long array of character names, it's better to have a list of something and refer to the count here instead of keeping track of a integer variable.